packages/dev/core/src/Physics/v2/Plugins/havokPlugin.ts

```diff
@@ -1221,6 +1221,50 @@ export class HavokPlugin implements IPhysicsEnginePluginV2 {
         }
     }
 
+    private _createOptionsFromGroundMesh(options: PhysicsShapeParameters) {
+        const mesh = options.groundMesh;
+        if (!mesh) {
+            return;
+        }
+        let pos = <FloatArray>mesh.getVerticesData(VertexBuffer.PositionKind);
+        const transform = mesh.computeWorldMatrix(true);
+        // convert rawVerts to object space
+        const transformedVertices: number[] = [];
+        let index: number;
+        for (index = 0; index < pos.length; index += 3) {
+            Vector3.TransformCoordinates(Vector3.FromArray(pos, index), transform).toArray(transformedVertices, index);

```

Comment:
This will generate a lot of vector3 objects. Especially if you convert to array, you can transform to a temp referenced vector

Reply:
Indeed! Fixed!